### PR TITLE
Replaced 'Exception Handling' with new 'Data Binding' chapter

### PR DESCRIPTION
**Pull request by chkal**
I finally found some time to finish my work on #101, which is about creating a new "Data Binding" chapter which replaces the old "Exception Handling" chapter.

The new chapter describes the MVC specific data binding handling, which can be enabled by placing a `@MvcBinding` annotation on your parameter binding.

The AsciiDoc source is a bit difficult to review, so I included a rendered version of the PDF which you can [download here](https://www.dropbox.com/s/x6aczpf9kckcfmh/jsr371-data-binding-chapter-2.pdf?dl=1). Please refer to pages 22-25 for your review.

Let me know what you think. There are a few details which haven't been discussed yet. Especially which data types an MVC implementation is required to support. So I just defined a ste of types for now. Feel free to comment on that if you think that something is missing.

I also included the TCK coverage report annotations. So this is done already and we can continue our work on the TCK soon.

Feel free to bring up anything that you don't like about this new chapter! Thanks!

### CODE REVIEW

File: spec/src/main/asciidoc/chapters/data-binding.asciidoc [hunk 1]
```diff
@@ -0,0 +1,165 @@
+[[data_binding]]
+== Data Binding
+
+This chapter discusses data binding in the MVC API. 
+Data binding is based on the underlying mechanism provided by JAX-RS, 
+but with additional support for I18N requirements and for handling data binding errors within the controller.
```
Comment (lefloh) [inline spec/src/main/asciidoc/chapters/data-binding.asciidoc]: Not to relevant but I think that `i18n` (lowercase) is more common.

File: spec/src/main/asciidoc/chapters/data-binding.asciidoc [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+    @POST
+    public String processForm() {
+        // ...
+    }
+}
+----
+
+Please note that usually `@MvcBinding` will be used with `@FormParam` bindings, as they are very common in web application.
+However, depending on the specific use case, it may also be useful to use it with `@QueryParam` and other binding annotations.
```
Comment (lefloh) [inline spec/src/main/asciidoc/chapters/data-binding.asciidoc]: Why the limitation to `@FormParam` at all? `/articles?maxPrice=500` is not a specific use case ;)

File: spec/src/main/asciidoc/chapters/data-binding.asciidoc (hunk 1 above)
Comment (chkal) [reply spec/src/main/asciidoc/chapters/data-binding.asciidoc]: Thanks, I just updated that.

File: spec/src/main/asciidoc/chapters/data-binding.asciidoc (hunk 2 above)
Comment (chkal) [reply spec/src/main/asciidoc/chapters/data-binding.asciidoc]: Good point. Back then I was actually not sure if people will user MVC binding with query parameter, but after thinking about this more, I guess it is a perfect use case. I adjusted the text accordingly.

**Comment by chkal** [comment]
Hey @lefloh,

thanks for your feedback. I updated a few parts of the document to address your concerns.

> One question: Should we really drop the whole ExceptionMapper part?

IMO there is no reason to keep the chapter. Actually exception handling is only mentioned, because validation and binding errors are described to be special exceptions that can be handled differently. However, I think that it is actually more something that belong to the data binding itself. The current implementation actually doesn't throw or catch exceptions any more, so the data binding errors are actually unrelated to the `ExceptionMapper` feature of JAX-RS. And the exception mapper section (old 3.1) is actually some kind of introduction for the following sections (3.2 and 3.3) which were integrated into the new data binding chapter.

I agree that there may be valid use cases for exception mappers in MVC apps. But actually there is nothing different from the core JAX-RS exception handling. And we also don't describe other JAX-RS features we are using (like matching URLs against resource paths) without modifying it in any way.

**Comment by lefloh** [comment]
Convinced! Thanks for the explanation!

**Pull request merged**